src/components/SettingsModal/SettingsModal.tsx

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import React from "react"
+import ReactDOM from "react-dom"
+import "./SettingsModal.scss"
+
+interface Props {
+  isActive: boolean
+  toggle: () => void
+}
+
+const SettingsModal: React.FC<Props> = ({ isActive, toggle }) => {
+  if (!isActive) {
+    return null
+  }
+
+  return ReactDOM.createPortal(
+    <React.Fragment>
+      <div className="modal is-active">
+        <div
+          role="button"
+          tabIndex={0}
+          className="modal-background"
+          onClick={toggle}
+        />
```

Comment:
The `eslint` error that is disabled by the comment at the top of the file can be fixed by adding 

```
onKeyPress={toggle}
```

More info on this can be found in the [a11y accessibility documentation](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/112261cbc84f5b7d74de9b427b529a10b41faece/docs/rules/click-events-have-key-events.md)